src/analysis/bounded_variation.lean

```diff
@@ -207,6 +206,34 @@ begin
   simp [u, edist_comm],
 end
 
+lemma lower_continuous_aux {ι : Type*} {F : ι → α → E} {p : filter ι}
+  {f : α → E} {s : set α} (Ffs : ∀ x ∈ s, filter.tendsto (λ i, F i x) p (𝓝 (f x)))
+  {v : ℝ≥0∞} (hv : v < evariation_on f s) : (∀ᶠ (n : ι) in p, v < evariation_on (F n) s) :=
+begin
+  obtain ⟨⟨n, ⟨u, um, us⟩⟩, hlt⟩ := lt_supr_iff.mp hv,
+  have : filter.tendsto (λ j, ∑ (i : ℕ) in finset.range n, edist (F j (u (i + 1))) (F j (u i)))
+           p (𝓝 (∑ (i : ℕ) in finset.range n, edist (f (u (i + 1))) (f (u i)))), by
+  { apply tendsto_finset_sum,
+    exact λ i hi, filter.tendsto.edist (Ffs (u i.succ) (us i.succ)) (Ffs (u i) (us i)) },
+  exact (eventually_gt_of_tendsto_gt hlt this).mono
+    (λ i h, lt_of_lt_of_le h (sum_le (F i) n um us)),
+end
+
+/--
+The map `λ f, evariation_on f s` is lower semicontinuous for pointwise convergence *on `s`*.
+Pointwise convergence on `s` is encoded here as uniform convergence on the family consisting of the
+singletons of elements of `s`.
+-/
+lemma lower_semicontinuous (s : set α) :
+  lower_semicontinuous (λ f : uniform_on_fun α E (s.image $ λ x, {x}), evariation_on f s) :=
+begin
+  intro f,
+  apply @lower_continuous_aux _ _ _ _ (uniform_on_fun α E (s.image singleton)) id (𝓝 f) f s _,
+  simpa only [uniform_on_fun.tendsto_iff_tendsto_uniformly_on, mem_image, forall_exists_index,
+             and_imp, forall_apply_eq_imp_iff₂,
+             tendsto_uniformly_on_singleton_iff_tendsto] using @filter.tendsto_id _ (𝓝 f),
+end
+
```

Comment:
can you also give the version for the uniform convergence topology?

Reply:
Is there a clean way to chain `uniform_on_fun α E {s}` being finer than `uniform_on_fun α E (s.image singleton)` and `lower_semicontinuous` staying continuous and reuse the `lower_semicontinuous` or is it better to just apply the `_aux` lemma?

Reply:
That's not clear to me. If the proof with the _aux lemma is just a few lines long, I'd probably go with it.

Reply:
I did a manual proof. It's a bit longer than the other one but I feel it's small enough not to bother with a high-level alternative.
It's also probably quite golfable.